engine/src/main/java/org/hibernate/search/engine/environment/classpath/spi/AggregatedClassLoader.java

```diff
@@ -22,7 +29,31 @@
 public final class AggregatedClassLoader extends ClassLoader {
 	private ClassLoader[] individualClassLoaders;
 
-	public AggregatedClassLoader(ClassLoader... classLoaders) {
+	public static AggregatedClassLoader createDefault() {
+		final LinkedHashSet<ClassLoader> orderedClassLoaderSet = new LinkedHashSet<>();
+
+		//  adding known class-loaders...
+		orderedClassLoaderSet.add( AggregatedClassLoader.class.getClassLoader() );
+
+		// then the TCCL, if one...
+		final ClassLoader tccl = locateTCCL();
+		if ( tccl != null ) {
+			orderedClassLoaderSet.add( tccl );
+		}
+
+		// finally the system classloader
+		final ClassLoader sysClassLoader = locateSystemClassLoader();
+		if ( sysClassLoader != null ) {
+			orderedClassLoaderSet.add( sysClassLoader );
+		}
+
+		// now build the aggregated class loader...
+		return new AggregatedClassLoader(
+				orderedClassLoaderSet.toArray( new ClassLoader[0] )
```

Comment:
Actually I'm not sure about, but maybe could have sense to pre-assign a size to the array.

Reply:
Passing `new ClassLoader[0]` is about as efficient as passing an array with the right size, at least it was last time I checked. I'll change that if it turns out there really is an efficiency problem :)